javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,244 @@
+  for (let i = 0; i < notes.length; i++) {
+    const lowerCaseContent = notes[i].content.toLowerCase();
+    if (lowerCaseContent.includes(keyword.toLowerCase())) {
+      console.log(`Notes: ${notes[i].content}`);
+    } else {
+      console.log(`No notes found`);
+    }
+  }
+}
+
+searchForNotes("laundry");
+
+//CactusIO-interactive (Smart phone usage app) optional
+
+const activities = [];
+const today = new Date(); //improve on date
+function addActivity(activity, duration) {
+  activities.push({
+    date: today.toLocaleDateString("en-US", {
+      year: "numeric",
+      month: "numeric",
+      day: "numeric",
+    }),
+    activity: activity,
+    duration: duration,
+  });
+
+  return activities;
+}
+
+console.log(addActivity("Youtube", 30));
+
+//Show my status
+const limitUsageTime = 120;
+function showStatus(activities, date) {
+  if (activities.length < 1) {
+    console.log(`Add some activities before calling showStatus`);
+  } else {
+    // let usageTime = 0;
+    // for (let i = 0; i < activities.length; i++) {
+    //   usageTime += activities[i].duration;
+    // }
+    // if (usageTime < limitUsageTime) {
+    //   console.log(
+    //     `You have added ${activities.length} activities. They amount to ${usageTime} min. of usage`
+    //   );
+    // } else {
+    //   console.log(`You have reached your limit, no more smartphoning for you!`);
+    // }
+    let numberOfActivities = 0;
+    for (let i = 0; i < activities.length; i++) {
+      if (date === activities[i].date) {
+        numberOfActivities += 1;
+      }
+      console.log(
+        `The number of activities on ${date} is ${numberOfActivities}`
+      );
+    }
+  }
+}
+
+showStatus(activities, "2/19/2025");
+
+// Extra feature
+function showActivities(activities) {
+  if (activities.length < 1) {
+    console.log(`You don't have any activities`);
+  } else {
+    for (let i = 0; i < activities.length; i++) {
+      console.log(
+        `On ${activities[i].date}, You have been on ${activities[i].activity} for ${activities[i].duration} minutes`
+      );
+    }
+  }
+}
+
+showActivities(activities);
+
+//Create a function for calculating the activity a user has spent the most time on.
+
+function activitySpentMostTimeOn(activities) {
+  if (activities.length < 1) {
+    console.log(`Please add some activities first`);
+  }
+
+  const activitiesTotalTime = {};
+
+  for (let i = 0; i < activities.length; i++) {
+    const activity = activities[i].activity;
+    const duration = activities[i].duration;
+
+    if (activitiesTotalTime[activity]) {
+      activitiesTotalTime[activity] += duration;
+    } else {
+      activitiesTotalTime[activity] = duration;
+    }
+  }
+
+  const allActivities = Object.keys(activitiesTotalTime);
+  const allActivitiesTime = Object.values(activitiesTotalTime);
+  let maxTime = allActivitiesTime[0];
+  let indexOfMaxTime = 0;
+  for (let i = 1; i < allActivitiesTime.length; i++) {
+    if (maxTime < activitiesTotalTime[i]) {
```

Comment:
```suggestion
    if (maxTime < allActivitiesTime[i]) {
```
Now the code is looking at the wrong list and if there is more than one activity it will not output the correct answer. With the right list he will find the maximum